geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateGatewaySenderCommand.java

```diff
@@ -149,9 +155,55 @@ public ResultModel createGatewaySender(
     ResultModel resultModel = ResultModel.createMemberStatusResult(gatewaySenderCreateResults);
     resultModel.setConfigObject(configuration);
 
+    if (!waitForGatewaySenderMBeanCreation(id, membersToCreateGatewaySenderOn)) {
+      resultModel.addInfo()
+          .addLine("Did not complete waiting for GatewaySenderMBean proxy creation");
+    }
+
     return resultModel;
   }
 
+  /*
+   * Wait for up tp 3 seconds for the proxy MBeans to be created.
+   */
+  @VisibleForTesting
+  boolean waitForGatewaySenderMBeanCreation(String id,
+      Set<DistributedMember> membersToCreateGatewaySenderOn) {
+    DistributedSystemMXBean dsMXBean = getManagementService().getDistributedSystemMXBean();
+    long startWaitTime = System.currentTimeMillis();
+
+    do {
+      try {
+        if (membersToCreateGatewaySenderOn.stream()
+            .allMatch(m -> gatewaySenderBeanExists(dsMXBean, m.getName(), id))) {
+          return true;
+        }
+
+        Thread.sleep(100);
```

Comment:
Can we move `GeodeAwaitility` to `geode-commons` so that we can use it here instead? It looks like it's helpful in the production code as well.

Reply:
Or `GeodeAwaitility` assumes some long timeouts for testing purpose, so maybe we should introduce at least `Awaitility` so that the defaults are more suitable for production.

Reply:
Ideally I'd prefer to have something that monitors the value and keeps extending the timeout while the value is increasing. I don't think it's worth adding `Awaitility` here.

Reply:
It's just that I have been told many many times that Thread.sleep(100) in our code is very very bad....

Reply:
Looks like these two blocks of code `waitForGatewaySenderMBeanCreation` and `waitForGatewaySenderMBeanDeletion` have similar constructs. We can put a general method in GfshCommand to do these sort of things like `waitFor("some lamda")`, that way, at least this "waiting" utility is in one place for all of our commands, and it's easy to switch the implementations later on.